src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt

```diff
@@ -40,8 +31,8 @@ val MessageGroup.parentEventId: String?
         }
     }
 
-val MessageGroup.allParentEventIds: Set<String>
-    get() = messagesList.mapNotNullTo(HashSet()) { anyMessage ->
+val MessageGroup.allParentEventIds: Set<String?>
+    get() = messagesList.mapTo(HashSet()) { anyMessage ->
         when {
             anyMessage.hasMessage() -> anyMessage.message.parentEventId.id.ifEmpty { null }
             anyMessage.hasRawMessage() -> anyMessage.rawMessage.parentEventId.id.ifEmpty { null }
```

Comment:
I think  `message.parentEventId.id.ifEmpty { null }` and `rawMessage.parentEventId.id.ifEmpty { null }` should be extracted into extension properties because you use same logic in `toErrorGroup` method